lib/plugin/instance.rb

```diff
@@ -721,6 +721,16 @@ def add_custom_reviewable_filter(filter)
     end
   end
 
+  # Receives a new mapping:
+  # 1. A symbol that represents the name of the resource.
+  # 2. A symbol that represents the name of the action.
+  # 3. An action that contains the controller mapping, optional params, and aliases.
+  def add_api_key_scope(resource, action_name, action)
+    reloadable_patch do
+      ApiKey.new_mapping(resource, action_name, action)
```

Comment:
We should make sure that these mappings get removed when plugins are disabled. The easiest way is probably to use a [`filtered_register` in the DiscoursePluginRegistry](https://github.com/discourse/discourse/blob/master/lib/discourse_plugin_registry.rb#L70-L78).